modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/ColocationGroup.java

```diff
@@ -216,7 +243,16 @@ public ColocationGroup finalizeMapping() {
             assignments.add(first != null ? Collections.singletonList(first) : Collections.emptyList());
         }
 
-        return new ColocationGroup(sourceIds, new ArrayList<>(nodes), assignments);
+        return new ColocationGroup(sourceIds, new ArrayList<>(nodes), assignments, cacheAssignment);
+    }
+
+    /** */
+    public ColocationGroup explicitMapping() {
```

Comment:
I'm not sure about the naming. We aren't going to explicitly map partitions and are about to send the request without it, right? Isn't it `implicitMapping` or even `skipPartitionsMapping()`?

Reply:
On the contrary, after `expicitMapping` call we are always sending assignment with the request (without this call we can skip assignment sending)